resources/build-npm.js

```diff
@@ -107,6 +129,44 @@ function buildPackageJSON() {
   delete packageJSON.scripts;
   delete packageJSON.devDependencies;
 
+  packageJSON.type = 'commonjs';
+
+  for (const entryPoint of entryPoints) {
+    if (!entryPoint.endsWith('index.ts')) {
+      continue;
+    }
+    const base = ('./' + path.dirname(entryPoint)).replace(/\/.?$/, '');
+    const generated = {};
+    generated[base] = {
+      types: {
+        import: base + '/index.js.d.mts',
+        default: base + '/index.d.ts',
+      },
+      /*
+       this is not automatically picked up by vitest, but we can instruct users to add it to their vitest config:
+      ```js title="vite.config.ts"
+      import { defineConfig } from 'vite';
+      export default defineConfig(async ({ mode }) => {
+        return {
+          resolve: mode === 'test' ? { conditions: ['vitest'] } : undefined,
+        };
+      });
+      ```
+       */
+      vitest: {
+        import: base + '/index.js.mjs',
+        default: base + '/index.js',
+      },
+      module: base + '/index.mjs',
+      import: base + '/index.js.mjs',
+      default: base + '/index.js',
+    };
```

Comment:
This hurts just a little :/

Reply:
I'm not sure it would help with the pain, but considering we have to teach vitest to use the ESM/CJS single package version anyway, perhaps the name might be better off more generic in case other tools need that as well?

And would it possibly help just to make the condition "node" => in your comment above, might vitest pick that up? Not sure about that, or whether it would interfere with the other runtimes....